src/javascript/JContent/ContentRoute/ToolBar/ViewModeSelector/ViewModeSelector.jsx

```diff
@@ -49,6 +49,8 @@ export const ViewModeSelector = ({selector, setTableViewModeAction}) => {
         availableModes = availableModes.filter(n => n !== PAGE_BUILDER);
     }
 
+    availableModes = availableModes.reduce((acc, val) => hiddenViewModes.includes(val) ? acc : acc.concat(val), []);
```

Comment:
reduce seems a bit awkward to use here. Maybe filter might be more appropriate:

```suggestion
    availableModes = availableModes.filter(vm => !hiddenViewModes.includes(vm));
```

Reply:
yeah sorry, I know I'm jumping the gun but wanted to catch while you're still editing away ;)